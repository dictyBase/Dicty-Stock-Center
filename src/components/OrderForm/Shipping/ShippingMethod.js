// @flow
import React, { useState } from "react"
import Grid from "@material-ui/core/Grid"
import RadioGroup from "@material-ui/core/RadioGroup"
import Radio from "@material-ui/core/Radio"
import FormControlLabel from "@material-ui/core/FormControlLabel"
import TextField from "../TextField"
import useStyles from "../formStyles"

type Props = {
  /** Function for handling radio button selection */
  handleChange: Function,
  /** Function to manually set Formik field values */
  setFieldValue: Function,
}

/**
 * ShippingMethod contains radio buttons and a text field for listing courier information.
 */

const ShippingMethod = (props: Props) => {
  const [shipAccountNum, setShipAccountNum] = useState(true)
  const [prepaidNotice, setPrepaidNotice] = useState(false)
  const classes = useStyles()
  const { handleChange, setFieldValue } = props

  const handleShipAccountChange = () => {
    setShipAccountNum(true)
    setPrepaidNotice(false)
  }

  const handlePrepaidLabelChange = () => {
    setShipAccountNum(false)
    setPrepaidNotice(true)
    setFieldValue("shippingAccountNumber", "sending prepaid shipping label")
  }

  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      className={classes.innerForm}>
      <Grid item xs={3}>
        <span className={classes.requiredText}>*</span> Shipping Account:
      </Grid>
      <Grid item xs={8}>
        <RadioGroup
          aria-label="Shipping Account"
          name="shippingAccount"
          onChange={handleChange}
          row>
          <FormControlLabel
            value="fedex"
            control={<Radio />}
            label="FedEx"
            onChange={handleShipAccountChange}
          />
          <FormControlLabel
            value="ups"
            control={<Radio />}
            label="UPS"
            onChange={handleShipAccountChange}
          />
          <FormControlLabel
            value="dhl"
            control={<Radio />}
            label="DHL"
            onChange={handleShipAccountChange}
          />
          <FormControlLabel
            value="prepaid"
            control={<Radio />}
            label="Send prepaid shipping label"
            onChange={handlePrepaidLabelChange}
          />
        </RadioGroup>
        {shipAccountNum && (
          <TextField
            type="text"
            name="shippingAccountNumber"
            placeholder="Shipping Account Number"
          />
        )}
        {prepaidNotice && (
          <div className={classes.panelBlue}>
            If using a prepaid shipping label, please send ASAP to{" "}
            <u>
              <a href="mailto:dictystocks@northwestern.edu" target="_top">
                dictystocks@northwestern.edu
              </a>
            </u>
          </div>
        )}
      </Grid>
    </Grid>
  )
}

export default ShippingMethod
