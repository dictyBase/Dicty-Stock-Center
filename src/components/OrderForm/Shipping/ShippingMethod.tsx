import React, { useState } from "react"
import Grid from "@material-ui/core/Grid"
import TextField from "../TextField"
import RequiredTextLabel from "../RequiredTextLabel"
import useStyles from "../formStyles"
import ShippingMethodPrepaidNotice from "./ShippingMethodPrepaidNotice"
import ShippingMethodRadioGroup from "./ShippingMethodRadioGroup"

type Props = {
  /** Function for handling radio button selection */
  handleChange: () => void
  /** Function to manually set Formik field values */
  setFieldValue: Function
}

/**
 * ShippingMethod contains radio buttons and a text field for listing courier information.
 */

const ShippingMethod = ({ handleChange, setFieldValue }: Props) => {
  const [shipAccountNum, setShipAccountNum] = useState(true)
  const [prepaidNotice, setPrepaidNotice] = useState(false)
  const classes = useStyles()

  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      className={classes.innerForm}>
      <RequiredTextLabel title="Shipping Account" />
      <Grid item xs={12} md={8}>
        <ShippingMethodRadioGroup
          setShipAccountNum={setShipAccountNum}
          setPrepaidNotice={setPrepaidNotice}
          handleChange={handleChange}
          setFieldValue={setFieldValue}
        />
        {shipAccountNum && (
          <TextField
            type="text"
            name="shippingAccountNumber"
            placeholder="Shipping Account Number"
          />
        )}
        {prepaidNotice && <ShippingMethodPrepaidNotice />}
      </Grid>
    </Grid>
  )
}

export default ShippingMethod
