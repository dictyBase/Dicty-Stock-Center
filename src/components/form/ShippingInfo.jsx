import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Flex, Box } from 'rebass'
import {
  RequiredText,
  FormGroup,
  FormControl,
  HelpBlock,
  ControlLabel,
  RadioInline
} from 'styles'

export default class ShippingInfo extends Component {
  displayName = 'form shipping information'

  static propTypes = {
      shipAccount: PropTypes.object.isRequired,
      shipAccountNum: PropTypes.object.isRequired
  }

  renderShipAccountNum = () => {
      const { shipAccountNum } = this.props
      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ 1 / 5 } />
          <Box w={ ['90%', '90%', 2 / 3] }>
            <FormControl
              {...shipAccountNum}
              placeholder="Shipping Account Number"
            />
            { shipAccountNum.error && (
              <HelpBlock>{ shipAccountNum.error }</HelpBlock>
            ) }
          </Box>
        </Flex>
      </FormGroup>
    )
  }

  render() {
      const { shipAccount } = this.props
      const hasError = shipAccount.touched && shipAccount.error
      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ 1 / 4 } ml={ 1 } mb={ 2 }>
            <ControlLabel>
              <RequiredText>* </RequiredText>
              Shipping Account:
            </ControlLabel>
          </Box>
          <Box w={ 2 / 3 } mr={ 2 } mb={ 1 }>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="Fedex"
                checked={ shipAccount.value === 'Fedex' }
              />&nbsp; FedEx
            </RadioInline>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="UPS"
                checked={ shipAccount.value === 'UPS' }
              />&nbsp; UPS
            </RadioInline>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="DHL"
                checked={ shipAccount.value === 'DHL' }
              />&nbsp; DHL
            </RadioInline>
            <RadioInline>
              <input
                type="radio"
                {...shipAccount}
                value="Will call 1-312-503-4169"
                checked={ shipAccount.value === 'Will call 1-312-503-4169' }
              />&nbsp; Call in Credit card # for FedEx Waybill 1-312-503-4169
            </RadioInline>
            { hasError && <HelpBlock>{ shipAccount.error }</HelpBlock> }
          </Box>
        </Flex>
        { !(shipAccount.value === 'Will call 1-312-503-4169') &&
          this.renderShipAccountNum() }
      </FormGroup>
    )
  }
}
