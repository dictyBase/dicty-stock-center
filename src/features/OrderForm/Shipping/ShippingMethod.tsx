import React, { useState } from "react"
import Typography from "@material-ui/core/Typography"
import Box from "@material-ui/core/Box"
import TextField from "../TextField"
import RequiredTextLabel from "../RequiredTextLabel"
import ShippingMethodPrepaidNotice from "./ShippingMethodPrepaidNotice"
import ShippingMethodRadioGroup from "./ShippingMethodRadioGroup"

/**
 * ShippingMethod contains radio buttons and a text field for listing courier information.
 */

const ShippingMethod = () => {
  const [shipAccountNum, setShipAccountNum] = useState(true)
  const [prepaidNotice, setPrepaidNotice] = useState(false)

  return (
    <Box mt={1} mb={2} p={2}>
      <RequiredTextLabel title="Shipping Account" variant="h3" />
      <Box mt={1} />
      <ShippingMethodRadioGroup
        setShipAccountNum={setShipAccountNum}
        setPrepaidNotice={setPrepaidNotice}
      />
      {shipAccountNum && (
        <TextField
          name="shippingAccountNumber"
          placeholder="Shipping Account Number"
        />
      )}
      {prepaidNotice && <ShippingMethodPrepaidNotice />}
      <Box mt={1} />
      <Typography component="p">
        <em>Note: credit card is not allowed for shipment</em>
      </Typography>
    </Box>
  )
}

export default ShippingMethod
