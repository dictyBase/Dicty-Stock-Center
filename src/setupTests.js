import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"

global.console = {
  log: jest.fn(),
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
