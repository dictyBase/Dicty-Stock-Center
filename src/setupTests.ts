import { configure } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-enzyme"
import { sprintf } from "sprintf-js"

let globalAny = global as any

globalAny.console = {
  log: (msg: string, args: any) => {
    if (typeof msg !== "string") {
      return
    }
    const str = sprintf(msg, args)
    process.stderr.write(str + "\n")
  },
  error: jest.fn(),
  warn: jest.fn(),
}

configure({ adapter: new Adapter() })
