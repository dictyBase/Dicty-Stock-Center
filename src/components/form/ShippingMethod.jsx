import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
    Panel,
    PanelHeader,
    PanelTitle,
    PanelBody
} from 'dicty-components-panel'
import ShippingInfo from './ShippingInfo'
import 'styles/core.scss'

// still need to pass in custom styles

export default class ShippingMethod extends Component {
    displayName = 'shipping method'

    static propTypes = {
        shipAccount: PropTypes.object.isRequired,
        shipAccountNum: PropTypes.object.isRequired,
        title: PropTypes.string.isRequired
    }

    render() {
        const { title, shipAccount, shipAccountNum } = this.props
        const panelStyle = { border: '1px solid #D2D7D3' }
        const headerStyle = {
            backgroundColor: '#337ab7',
            padding: '20px',
            borderColor: '#4B77BE'
        }
        const titleStyle = {
            color: '#ffffff',
            fontWeight: '200',
            fontSize: '20px'
        }
        return (
            <Panel collapse style={ panelStyle }>
                <PanelHeader style={ headerStyle }>
                    <PanelTitle style={ titleStyle }>{ title }</PanelTitle>
                </PanelHeader>
                <PanelBody>
                    <ShippingInfo
                        shipAccount={ shipAccount }
                        shipAccountNum={ shipAccountNum }
                    />
                </PanelBody>
            </Panel>
        )
    }
}
