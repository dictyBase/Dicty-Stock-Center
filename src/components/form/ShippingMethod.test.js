import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import "../../setupTests"
import ShippingMethod from "./ShippingMethod"
import ShippingInfo from "./ShippingInfo"
import { shallow } from "enzyme"

describe("form/ShippingMethod", () => {
  const shipAccount = { value: "ups", touched: true }
  const shipAccountNum = { value: "6262", touched: true }
  const title = "Method"

  const wrapper = shallow(
    <ShippingMethod
      shipAccount={shipAccount}
      shipAccountNum={shipAccountNum}
      title={title}
    />,
  )

  it("should render <ShippingInfo>", () => {
    expect(
      wrapper.contains(
        <ShippingInfo
          shipAccount={shipAccount}
          shipAccountNum={shipAccountNum}
        />,
      ),
    ).toEqual(true)
  })
})

test("matching a snapshot of ShippingMethod", () => {
  const shipAccount = {
    value: "Fedex",
    touched: true,
  }
  const shipAccountNum = {
    value: "",
    touched: false,
  }
  const title = "Method"

  const component = renderer.create(
    <ShippingMethod
      shipAccount={shipAccount}
      shipAccountNum={shipAccountNum}
      title={title}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
