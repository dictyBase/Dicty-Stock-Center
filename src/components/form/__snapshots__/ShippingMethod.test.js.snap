// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ShippingMethod 1`] = `
.c0 {
  margin-top: 5px;
  border: 1px solid;
  box-shadow: 0 1px 1px rgba(0,0,0,0.05);
  font-size: 14px;
  line-height: 1.4285714;
  color: #333;
  border-radius: 4px;
  font-family: Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c1 {
  background-color: #15317e;
  border-color: #ddd;
  padding: 10px 15px;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  font-family: Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.42857143;
  cursor: pointer;
}

.c2 {
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  color: white;
  font-family: inherit;
  font-weight: 500;
  line-height: 1.1;
}

.c3 {
  background-color: transparent;
  color: white;
  cursor: pointer;
}

.c4 {
  padding: 0px 15px;
  font-family: Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.42857143;
  border-top: 1px solid;
  border-top-width: 1px;
  -webkit-transition: height 600ms cubic-bezier(0.23,1,0.32,1),border-top-width 400ms cubic-bezier(0.23,1,0.32,1);
  transition: height 600ms cubic-bezier(0.23,1,0.32,1),border-top-width 400ms cubic-bezier(0.23,1,0.32,1);
  height: auto;
  overflow: hidden;
}

.c5 {
  height: 15px;
}

.c7 {
  display: inline-block;
  padding-top: 7px;
  margin-bottom: 0;
  text-align: right;
  font-weight: 600;
}

.c6 {
  margin-right: -15px;
  margin-left: -15px;
}

.c10 {
  display: block;
  width: 100%;
  height: 34px;
  padding: 6px 12px;
  font-size: 14px;
  line-height: 1.42857143;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075);
  -webkit-transition: border-color ease-in-out 0.15s, -webkit-box-shadow ease-in-out 0.15s;
  -o-transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s;
  -webkit-transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s;
  transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s;
}

.c10:focus {
  border-color: #66afe9;
  outline: 0;
  -webkit-box-shadow: inset 0 1px 1px rgba(0,0,0,0.075), 0 0 8px rgba(102,175,233,0.6);
  box-shadow: inset 0 1px 1px rgba(0,0,0,0.075), 0 0 8px rgba(102,175,233,0.6);
}

.c9 {
  position: relative;
  display: inline-block;
  padding-right: 20px;
  margin-bottom: 0;
  font-weight: normal;
  vertical-align: middle;
  cursor: pointer;
}

.c8 {
  color: #dc3545;
}

<div
  className="c0"
  style={
    Object {
      "border": "1px solid #D2D7D3",
      "marginBottom": 0,
    }
  }
>
  <div
    className="c1"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#337ab7",
        "borderBottom": "0px none #ddd",
        "borderColor": "#4B77BE",
        "padding": "20px",
      }
    }
  >
    <h3
      className="c2"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "20px",
          "fontWeight": "200",
        }
      }
    >
      <div
        className="c3"
        onClick={[Function]}
      >
        Method
      </div>
    </h3>
  </div>
  <div
    className="panel-body c4"
    open={true}
    style={Object {}}
  >
    <div
      className="c5"
    />
    <div
      className="c6"
    >
      <div
        className="eDtRyC sc-bdVaJa iHZvIS"
        is={null}
      >
        <div
          className="sc-bdVaJa cyncUw"
          is={null}
        >
          <label
            className="c7"
          >
            <span
              className="c8"
            >
              * 
            </span>
            Shipping Account:
          </label>
        </div>
        <div
          className="sc-bdVaJa XupaC"
          is={null}
        >
          <label
            className="c9"
          >
            <input
              checked={true}
              touched={true}
              type="radio"
              value="Fedex"
            />
              FedEx
          </label>
          <label
            className="c9"
          >
            <input
              checked={false}
              touched={true}
              type="radio"
              value="UPS"
            />
              UPS
          </label>
          <label
            className="c9"
          >
            <input
              checked={false}
              touched={true}
              type="radio"
              value="DHL"
            />
              DHL
          </label>
          <label
            className="c9"
          >
            <input
              checked={false}
              touched={true}
              type="radio"
              value="Will call 1-312-503-4169"
            />
              Call in Credit card # for FedEx Waybill 1-312-503-4169
          </label>
        </div>
      </div>
      <div
        className="c6"
      >
        <div
          className="eDtRyC sc-bdVaJa iHZvIS"
          is={null}
        >
          <div
            className="sc-bdVaJa cyncUw"
            is={null}
          />
          <div
            className="sc-bdVaJa cjDWjg"
            is={null}
          >
            <input
              className="c10"
              placeholder="Shipping Account Number"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="c5"
    />
  </div>
</div>
`;
